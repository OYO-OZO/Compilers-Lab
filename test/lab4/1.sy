int a;

int main()
{
    int a,f,h,o;
    int s[30];
    a = 1 + 2;
    f = 3.23 * 3.5E-10;
    h = 0x2f;
    o = 035;
    a++;
    if(a < 5)
        return 1;
    return 0;
}