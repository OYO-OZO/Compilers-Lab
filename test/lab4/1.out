[DEBUG LAB4]: 	INT		int	1	0
[DEBUG LAB4]: 	ID		a	1	4		1
[DEBUG LAB4]: 	SEMI		;	1	5
[DEBUG LAB4]: 	INT		int	3	0
[DEBUG LAB4]: 	ID		main	3	4		2
[DEBUG LAB4]: 	LP		(	3	8
[DEBUG LAB4]: 	RP		)	3	9
[DEBUG LAB4]: 	LC		{	4	0
[DEBUG LAB4]: 	INT		int	5	4
[DEBUG LAB4]: 	ID		a	5	8		3
[DEBUG LAB4]: 	COMMA		,	5	9
[DEBUG LAB4]: 	ID		f	5	10		4
[DEBUG LAB4]: 	COMMA		,	5	11
[DEBUG LAB4]: 	ID		h	5	12		5
[DEBUG LAB4]: 	COMMA		,	5	13
[DEBUG LAB4]: 	ID		o	5	14		6
[DEBUG LAB4]: 	SEMI		;	5	15
[DEBUG LAB4]: 	INT		int	6	4
[DEBUG LAB4]: 	ID		s	6	8		7
[DEBUG LAB4]: 	LB		[	6	9
[DEBUG LAB4]: 	NUMBER		30	6	10
[DEBUG LAB4]: 	RB		]	6	12
[DEBUG LAB4]: 	SEMI		;	6	13
[DEBUG LAB4]: 	ID		a	7	4		3
[DEBUG LAB4]: 	ASSIGN		=	7	6
[DEBUG LAB4]: 	NUMBER		1	7	8
[DEBUG LAB4]: 	ADD		+	7	10
[DEBUG LAB4]: 	NUMBER		2	7	12
[DEBUG LAB4]: 	SEMI		;	7	13
[DEBUG LAB4]: 	ID		f	8	4		4
[DEBUG LAB4]: 	ASSIGN		=	8	6
[DEBUG LAB4]: 	FLOATNUMBER		3.23	8	8
[DEBUG LAB4]: 	MUL		*	8	13
[DEBUG LAB4]: 	FLOATNUMBER		3.5E-10	8	15
[DEBUG LAB4]: 	SEMI		;	8	22
[DEBUG LAB4]: 	ID		h	9	4		5
[DEBUG LAB4]: 	ASSIGN		=	9	6
[DEBUG LAB4]: 	NUMBER	9	8	47
[DEBUG LAB4]: 	SEMI		;	9	12
[DEBUG LAB4]: 	ID		o	10	4		6
[DEBUG LAB4]: 	ASSIGN		=	10	6
[DEBUG LAB4]: 	NUMBER	10	8	NUMBER	29
[DEBUG LAB4]: 	SEMI		;	10	11
[DEBUG LAB4]: 	ID		a	11	4		3
[DEBUG LAB4]: 	INC		++	11	5
[DEBUG LAB4]: 	SEMI		;	11	7
[DEBUG LAB4]: 	IF		if	12	4
[DEBUG LAB4]: 	LP		(	12	6
[DEBUG LAB4]: 	ID		a	12	7		3
[DEBUG LAB4]: 	LESS		<	12	9
[DEBUG LAB4]: 	NUMBER		5	12	11
[DEBUG LAB4]: 	RP		)	12	12
[DEBUG LAB4]: 	RETURN		return	13	8
[DEBUG LAB4]: 	NUMBER		1	13	15
[DEBUG LAB4]: 	SEMI		;	13	16
[DEBUG LAB4]: 	RETURN		return	14	4
[DEBUG LAB4]: 	NUMBER		0	14	11
[DEBUG LAB4]: 	SEMI		;	14	12
[DEBUG LAB4]: 	RC		}	15	0
